node_package/src/clientStartup.ts

```diff
@@ -150,7 +152,8 @@ function render(el: Element, railsContext: RailsContext): void {
       }
 
       // Hydrate if available and was server rendered
-      const shouldHydrate = !!ReactDOM.hydrate && !!domNode.innerHTML;
+      // @ts-expect-error potentially present if React 18 or greater
```

Comment:
Please check if the `@ts-expect-error` messages are still needed.

Reply:
Unfortunately, these are still needed.  We could upgrade the types library to the react beta to get around having to do this.

Reply:
Pros and cons of doing that?

Reply:
I think the pros are:

1. Using an API that is up-to-date with the latest React.

The cons are a couple:

1. It would be depending on beta code for our types
2. It could potentially have removed deprecated types that are no longer used in React 18.  This could be seen as a positive as well because it would alert us to changes in the API that we might be using.

My recommendation would be to ignore it for now until it gets out of beta and then upgrade the types.